### PR TITLE
[GEOS-7150] Features counted twice for WFS queries with GeoJSON responses

### PR DESCRIPTION
**Pull request by aaime**

### CODE REVIEW

File: src/wfs/src/main/java/org/geoserver/wfs/CountExecutor.java [hunk 1]
```diff
@@ -40,7 +40,8 @@ public int getCount() throws IOException {
         if(providedCount != COUNT_UNSET) {
             return providedCount;
         } else {
-            return source.getCount(query);
+            // make sure we get a count, feature source can return -1
+            return source.getFeatures(query).size();
```
Comment (jodygarnett) [inline src/wfs/src/main/java/org/geoserver/wfs/CountExecutor.java]: not sure if the comment goes with the code here? Do you:

````
// mean make sure we get a count, source.getCount(query) can return -1
```

File: src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java [hunk 2]
```diff
@@ -296,7 +298,7 @@ public FeatureCollectionResponse run(GetFeatureRequest request)
 
         boolean isNumberMatchedSkipped = false;
         int count = 0; // should probably be long
-        int totalCount = 0;
+        BigInteger totalCount = BigInteger.ZERO;
```
Comment (jodygarnett) [inline src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java]: BigInteger really?

File: src/wfs/src/main/java/org/geoserver/wfs/CountExecutor.java (hunk 1 above)
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/CountExecutor.java]: Yes, that is the idea.

File: src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java (hunk 2 above)
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java]: That's what the EMF interface takes, complain with the EMF code generator, not me :)

File: src/wfs/pom.xml [hunk 3]
```diff
@@ -42,6 +42,10 @@
       <artifactId>gt-shapefile</artifactId>
     </dependency>
     <dependency>
+      <groupId>cglib</groupId>
```
Comment (aaime) [inline src/wfs/pom.xml]: Verified, it's indeed not going to add a new jar in the GeoServer release:

````
aaime@hydra ~/devel/gs_releases/geoserver-2.7.2/webapps/geoserver/WEB-INF/lib $ ls *cglib*
   cglib-nodep-2.2.jar
````

File: src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONGetFeatureResponse.java [hunk 4]
```diff
@@ -130,20 +130,10 @@ protected void write(FeatureCollectionResponse featureCollection, OutputStream o
         boolean hasGeom = false;
 
         // get feature count for request
-        BigInteger featureCount = null;
-        // for WFS 1.0.0 and WFS 1.1.0 a request with the query must be executed
-        if(describeFeatureType != null) {
-            if (describeFeatureType.getParameters()[0] instanceof GetFeatureType) {
-                featureCount = BigInteger.valueOf(getFeatureCountFromWFS11Request(describeFeatureType, wfs));
```
Comment (sampov2) [inline src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONGetFeatureResponse.java]: I think we could get rid of getFeatureCountFromWFS11Request() as well

**Pull request merged**